### DISCUSSION TITLE
notification when the focused monitor switches?

### DISCUSSION THREAD

**Discussion by yayuwang0624** [bakkeby/dwm-flexipatch#376] [Q&A]
I want an explicit notification (e.g. notify-send) when the focused monitor changes. I tried to add an extra SHCMD associated with the `focusmon` function so it works when I manually switch the monitor, but it doesn't work when `_NET_ACTIVE_WINDOW` changes the monitor.

Is there any means to achieve that?

**Comment by bakkeby** [comment]
I checked and it looks like it is possible to achieve this by using the [IPC](https://dwm.suckless.org/patches/ipc/) patch and subscribing to `monitor_focus_change_event`.

```sh
$ dwm-msg subscribe monitor_focus_change_event
{"result":"success"}
{
    "monitor_focus_change_event": {
        "old_monitor_number": 0,
        "new_monitor_number": 1
    }
}

{
    "monitor_focus_change_event": {
        "old_monitor_number": 1,
        "new_monitor_number": 0
    }
}

{
    "monitor_focus_change_event": {
        "old_monitor_number": 0,
        "new_monitor_number": 1
    }
}

{
    "monitor_focus_change_event": {
        "old_monitor_number": 1,
        "new_monitor_number": 0
    }
}
```

It would be possible to have this drive the script that sends the notifications.
